docs/rules/no-unused-expressions.md

```diff
@@ -39,6 +39,21 @@ a && b()
 a, b()
 
 c = a, b;
+
+a() && function namedFunctionInExpressionContext () {f();}
+
+(function anIncompleteIIFE () {});
+
```

Comment:
From your comments in #5446, my understanding is that if a rule has options, there should still be a separate level 3 heading for the default option(s) showing correct and incorrect code.

If that is the case, then my feeling is that "Rule Details" should be reserved for a more general introduction to the rule, rather than a discussion of edge cases.

Why? Because I think it can be helpful to be able to skip directly to the examples to induce the rule details, rather than have to read through an abstract discussion unless that is desired.

So, in the case of the discussion I'm adding as far as the treatment of strings re: directive prologue, although it begins in prose rather than code sample, I think that it is more suitable for this discussion to remain between the incorrect and correct rules since the discussion describes specifically when string expressions will be correct or not. Normally, I think only code examples should be needed here, but I think in this case, since it is more about position, we can't put this along with the other examples without confusing things, so I think a little description within the examples section is merited (though I think it wouldn't hurt to duplicate a little here, and *also* add your more generic wording directly under heading 2 of Rule Details after the note about sequence expressions that "This rule does not apply to directives (which are in the form of literal string expressions at the beginning of a script, module, or function).").

The reason I joined the directives into one "code well" was to draw out the fact that, so long as they are not interrupted by other types of statements, such multiple strings can all be treated as part of the prologue.